config/config_params.go

```diff
@@ -317,6 +317,7 @@ func (config *Config) DatastoreConfig() api.CalicoAPIConfig {
 		if err != nil {
```

Comment:
Hmm.. this code is assuming that `DATASTORE_TYPE=kubernetes` is set.  I suppose the issue is that `config.DatastoreType` can be `kubernetes` without that environment variable set.

Changing the `cfg.Spec.DatastoreType` below is only half of the problem - there's still no way to configure things like where the k8s API is from say, a Felix config file.